### ISSUE TITLE
Input required and not supplied: gpg_private_key

### ISSUE THREAD

**Issue by Jad-Jbara** [crazy-max/ghaction-import-gpg#172]
### Behaviour
Importing GPG keys to decrypt file, triggering error `Input required and not supplied` causing the job to fail.

I tried changing the workflow from `pull_request` to `pull_request_target` cause as far as I understood, Github changed permissions to PR's so that secrets wouldn't be accessed for security reasons.

I also tried multiple things but nothing works. (other secrets are accessed properly and other steps were working fine, it's just the ones related to the GPG secrets.
 
#### Steps to reproduce this issue

1. Add keys as Environment secrets
2. Run Job on creating PR

#### Expected behaviour

It should read keys from Secrets

#### Actual behaviour

Job fails with error: Input required and not supplied: gpg_private_key

### Configuration

* Repository URL (if public):  Private Repo
* Build URL (if public): Private Repo

```yml

name: Build APK and push to S3
on: pull_request_target


jobs:
  setup:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        # os: [ubuntu-latest, macos-latest, windows-latest]
        os: [ubuntu-latest]
        node: [14] # [10, 12]

    permissions:
      pull-requests: write
      contents: write

    steps:
      - uses: actions/setup-node@v3
        with:
          node-version: ${{ matrix.node }}

      - name: checkout
        uses: actions/checkout@v3
        with:
          fetch-depth: 0 # All history

      - name: cache node_modules
        id: node_modules_cache_id
        uses: actions/cache@v1
        with:
          path: node_modules
          key: ${{ matrix.os }}-node-v${{ matrix.node }}-deps-${{ hashFiles(format('{0}{1}', github.workspace, '/yarn.lock')) }}

      - name: install
        run: yarn --check-files --frozen-lockfile --non-interactive

      - name: cache workspace
        uses: actions/cache@v3
        with:
          path: ${{ github.workspace }}
          key: ${{ matrix.os }}-node-v${{ matrix.node }}-react-native-${{ github.sha }}

      - name: Import GPG Key
        uses: crazy-max/ghaction-import-gpg@v5
        with:    
          gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
          passphrase: ${{ secrets.GPG_PASSPHRASE }}
        # env:
        #   gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
        #   passphrase: ${{ secrets.GPG_PASSPHRASE }}

      - name: List keys
        run: gpg -K

```

### Logs

[logs_33.zip](https://github.com/crazy-max/ghaction-import-gpg/files/11562369/logs_33.zip)

**Comment by BeneHa** [comment]
Not sure if it is the same for you but I had the same issue, this fixed it for me:
- In the parent pipeline that is calling that one, use "secrets: inherit" so the secrets actually get passed down to this one
- In the import step, pass in the secrets via the "env" option

**Comment by crazy-max** [comment]
Sorry I don't bring support directly related to GitHub Actions. I suggest to check the official [documentation](https://docs.github.com/actions) or ask in https://github.com/orgs/community/discussions/categories/actions.

**Comment by wesley-dean-flexion** [comment]
@Jad-Jbara , I found this issue after running into this exact problem.  Per @BeneHa 's second point, I was able to resolve the issue by bringing in the private key and its passphrase into the environment.

https://github.com/wesley-dean/routeros_ssl/blob/main/.github/workflows/megalinter.yml#L30-L32

What's interesting to me is that on another project, I use this exact GitHub Action to sign commits and it works just fine.  The only difference between the two appears to be that the other project belongs to an organization and the one that was causing the errors did not.  (per @BeneHa 's first point).
